static/js/controllers/inbox.js

```diff
@@ -202,7 +212,11 @@ require('moment/locales');
         });
       };
 
-      $scope.$on('ContactUpdated', function() {
+      $scope.$on('ContactUpdated', function(contact) {
+        if (!contact || contact._deleted) {
+          // allow the ContactCtrl to handle contact deletion
+          return;
+        }
```

Comment:
Sure a small window. We could implement a request cancellation system where if a request is made through the HttpWrapper which is identical to one already sent then the first one gets cancelled? Low priority as it's unlikely to happen often.